nucypher/blockchain/eth/providers.py

```diff
@@ -31,13 +32,23 @@ def _get_websocket_provider(provider_uri):
 
 def _get_infura_provider(provider_uri):
     # https://web3py.readthedocs.io/en/latest/providers.html#infura-mainnet
-    infura_envvar = 'WEB3_INFURA_API_SECRET'
-    if infura_envvar not in os.environ:
-        raise ProviderError(f'{infura_envvar} must be set in order to use an Infura Web3 provider.')
-    from web3.auto.infura import w3
+
+    uri_breakdown = urlparse(provider_uri)
+    infura_envvar = 'WEB3_INFURA_PROJECT_ID'
+    os.environ[infura_envvar] = os.environ.get(infura_envvar, uri_breakdown.netloc)
+
+    try:
+        # TODO: Only testnet for now
+        from web3.auto.infura.goerli import w3
+
+    except InfuraKeyNotFound:
+        raise ProviderError(f'{infura_envvar} must be provided in order to use an Infura Web3 provider.')
```

Comment:
same comment as below about uri.

```suggestion
        raise ProviderError(f'{infura_envvar} must be provided in order to use an Infura Web3 provider {provider_uri}.')
```

Reply:
:heavy_check_mark: